sanic/asgi.py

```diff
@@ -164,10 +169,14 @@ async def __aiter__(self):
                 yield data
 
     def respond(self, response):
+        if self.stage is not Stage.HANDLER:
+            self.stage = Stage.FAILED
+            raise RuntimeError("Response already started")
         response.stream, self.response = self, response
         return response
```

Comment:
Disconnect an earlier response here as well, to stay compatible with http.py.

Reply:
Fixed